app/logic/emailHandler.py

```diff
@@ -187,3 +189,11 @@ def update_email_template(self):
         except Exception as e:
             print("Error updating email template record: ", e)
             return False
+
+    def retrieve_last_email(event_id):
+        try:
+            get_query = EmailLog.select().where(EmailLog.event==event_id).order_by(EmailLog.dateSent.desc()).get()
+            return get_query
+        except Exception as e:

```

Comment:
If we are using exceptions as a normal condition we have to only treat the expected exception as normal. Catching ALL exceptions means if something goes wrong with our application it will just print "No last email found" and continue on.